### PR TITLE
feat: geo-blocking, replace `gorgon` with `utils-rs`

### PR DESCRIPTION
**Pull request by xav**
# Description

Integrates the geoblocking feature from utils-rs. Currently blocked countries: KP, IR, CU, SY.

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/handlers/notify.rs [hunk 1]
```diff
@@ -147,18 +147,16 @@ async fn process_publish_jobs(
...
-            .lookup_geo_data(addr.ip())
-            .map_or((None, None, None), |geo| {
-                (
-                    geo.country,
-                    geo.continent,
-                    geo.region.map(|r| Arc::from(r.join(", "))),
-                )
-            });
+    let geo_ip = state
+        .analytics
```
Comment (chris13524) [inline src/handlers/notify.rs:151]: Should this be removed from `analytics` struct?

File: src/lib.rs [hunk 2]
```diff
@@ -199,3 +210,65 @@ fn create_http_client(
...
+    let aws_config = match &config.s3_endpoint {
+        Some(s3_endpoint) => {
+            info!(%s3_endpoint, "initializing analytics with custom s3 endpoint");
+
+            aws_sdk_s3::config::Builder::from(&shared_config)
+                .endpoint_url(s3_endpoint)
+                .build()
+        }
+        _ => aws_sdk_s3::config::Builder::from(&shared_config).build(),
+    };
```
Comment (chris13524) [inline src/lib.rs:246]:
```suggestion
    let mut builder = aws_sdk_s3::config::Builder::from(&shared_config);
    if let Some(s3_endpoint) = &config.s3_endpoint {
        info!(%s3_endpoint, "initializing analytics with custom s3 endpoint");
        builder = builder.endpoint_url(s3_endpoint);
    };
    aws_config = builder.build();
```

**Review by chris13524 — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (xav) [reply src/lib.rs:246]: Same result, I do like the match

Comment (xDarksome) [reply src/lib.rs:246]: Seems like everyone has an opinion on this across services :smile: 
Another reason to have it in `utils-rs`

I do like the @chris13524 's version more though

Ideally, the builder should have an API that supports both `T` and `Option<T>` for each configuration option. But for some reason no one wants to _build their builders_ in such a way

Comment (xav) [reply src/lib.rs:246]: It's a bit like traits and Arcs/Refs, easy to forget and kinda stupid that there's no easy  way to say "that works for refs too, trust me"

Comment (xav) [reply src/lib.rs:246]: As Ivan said, let's stop bikeshedding and roll with that, then push it to `utils-rs` later.

**Pull request merged**